src/routes/onboarding/onboarding.tsx

```diff
@@ -0,0 +1,365 @@
+  }
+};
+
+export const Onboarding = () => {
+  const [form, setForm] = useState({
+    jiraUrl: ifDev("JIRA_BASE_URL") as URL | null,
+    base64Token: ifDev("JIRA_API_KEY") as string,
+    boardId: ifDev("JIRA_BOARD_ID") as string,
+  });
+
+  const [profile, setProfile] = useState<JiraProfile | null>(null);
+
+  const requester = useRequester(form.jiraUrl, form.base64Token);
+
+  const [step, setStep] = useState<OnboardingStep>(OnboardingStep.Welcome);
+
+  const Step = {
+    [OnboardingStep.Welcome]: () => (
+      <WelcomeStep onNext={() => setStep(OnboardingStep.JiraUrl)} />
+    ),
+    [OnboardingStep.JiraUrl]: () => (
+      <JiraUrlStep
+        initialValue={form.jiraUrl}
+        onSubmit={(jiraUrl) => {
+          setForm((prev) => ({ ...prev, jiraUrl }));
+          setStep(OnboardingStep.Credentials);
+        }}
+      />
+    ),
+    [OnboardingStep.Credentials]: () => {
+      const decodedToken = form.base64Token
+        ? Buffer.from(form.base64Token, "base64").toString("utf8")
+        : null;
+
+      const [email, apiToken] = decodedToken?.split(":") ?? [];
+
+      return (
+        <CredentialsStep
+          email={email}
+          apiToken={apiToken}
+          onSubmit={({ email, apiToken }) => {
+            const base64Token = Buffer.from(`${email}:${apiToken}`).toString(
+              "base64",
+            );
+
+            setForm((prev) => ({ ...prev, email, base64Token }));
+
+            setStep(OnboardingStep.ConfirmCredentials);
+          }}
+        />
+      );
+    },
+    [OnboardingStep.ConfirmCredentials]: () => (
+      <CredentialConfirmStep
+        request={requester!}
+        onConfirm={(profile) => {
+          setProfile(profile);
+
+          setStep(OnboardingStep.BoardSelection);
+        }}
+        onCancel={() => setStep(OnboardingStep.Credentials)}
+      />
+    ),
+    [OnboardingStep.BoardSelection]: () => (
+      <BoardSelectionStep
+        request={requester!}
+        onSubmit={(boardId) => {
+          setForm((prev) => ({ ...prev, boardId }));
+
+          setStep(OnboardingStep.Finish);
+        }}
+      />
+    ),
+    [OnboardingStep.Finish]: () => (
+      <FinishStep
+        data={asNonNullable({
+          jiraUrl: form.jiraUrl,
+          apiToken: form.base64Token,
+          boardId: form.boardId,
+          profile,
+        })}
+      />
+    ),
+  }[step];
+
+  return (
+    <Box
+      justifyContent="center"
+      alignItems="center"
+      width="100%"
+      height="100%"
+      flexDirection="column"
+    >
+      <Box height="auto" width="50%">
+        <Box
+          flexDirection="column"
+          justifyContent="center"
+          alignItems="center"
+          borderColor="blue"
+          borderStyle="round"
+          padding={1}
+          gap={2}
+        >
+          {step !== OnboardingStep.Finish && <Text>Welcome to {APP_NAME}</Text>}
+
+          <Box minWidth="25%" justifyContent="center">
+            <Step />
+          </Box>
```

Comment:
TODO: not all steps have the same width, looks a bit weird having it jump around in size